main/adapter-commons/src/service.ts

```diff
@@ -1,84 +1,206 @@
+   * Sanitize the incoming data, e.g. removing invalid keywords etc.
+   *
+   * @param data The data to sanitize
+   * @param _params Service call parameters
+   * @returns The sanitized data
+   */
+  async sanitizeData<X = Partial<D>>(data: X, _params?: P) {
+    return data;
+  }
+
+  /**
+   * Returns a sanitized version of `params.query`, converting filter values
+   * (like $limit and $skip) into the expected type. Will throw an error if
+   * a `$` prefixed filter or operator value that is not allowed in `filters`
+   * or `operators` is encountered.
+   *
+   * @param params The service call parameter.
+   * @returns A new object containing the sanitized query.
+   */
+  async sanitizeQuery(params: P = {} as P): Promise<Query> {
+    const options = this.getOptions(params);
+    const { query, filters } = filterQuery(params.query!, options);
+
+    return {
+      ...filters,
+      ...query,
+    };
+  }
+
+  abstract $find(
+    _params?: P & { paginate?: PaginationOptions }
+  ): Promise<Paginated<T>>;
+  abstract $find(_params?: P & { paginate: false }): Promise<T[]>;
+  abstract $find(params?: P): Promise<T[] | Paginated<T>>;
+
+  /**
+   * Retrieve all resources from this service, skipping any service-level hooks but sanitize the query
+   * with allowed filters and properties by calling `sanitizeQuery`.
    *
    * @param params - Service call parameters {@link Params}
    * @see {@link HookLessServiceMethods}
    * @see {@link https://docs.feathersjs.com/api/services.html#find-params|Feathers API Documentation: .find(params)}
    */
-  _find (params?: AdapterParams): Promise<T | T[] | Paginated<T>>;
+  async _find(
+    _params?: P & { paginate?: PaginationOptions }
+  ): Promise<Paginated<T>>;
+  async _find(_params?: P & { paginate: false }): Promise<T[]>;
+  async _find(params?: P): Promise<T | T[] | Paginated<T>>;
+  async _find(params?: P): Promise<T | T[] | Paginated<T>> {
+    const query = await this.sanitizeQuery(params);
+
+    // @ts-ignore<typescript overload problem>
+    return this.$find({
+      ...params,
+      query,
+    } as P & { paginate?: PaginationOptions });
+  }
+
+  abstract $get(id: Id, params?: P): Promise<T>;
 
   /**
-   * Retrieve a single resource matching the given ID, skipping any service-level hooks.
+   * Retrieve a single resource matching the given ID, skipping any service-level hooks but sanitize the query
+   * with allowed filters and properties by calling `sanitizeQuery`.
    *
    * @param id - ID of the resource to locate
    * @param params - Service call parameters {@link Params}
    * @see {@link HookLessServiceMethods}
    * @see {@link https://docs.feathersjs.com/api/services.html#get-id-params|Feathers API Documentation: .get(id, params)}
    */
-  _get (id: Id, params?: AdapterParams): Promise<T>;
+  async _get(id: Id, params?: P): Promise<T> {
+    const query = await this.sanitizeQuery(params);
+
+    return this.$get(id, {
+      ...params,
+      query,
+    } as P);
```

Comment:
Here I had to cast the typ as `P`, because TypeScript complained about the following

> Argument of type '{ query: Query; adapter?: Partial<AdapterServiceOptions> | undefined; paginate?: PaginationParams | undefined; provider?: string | undefined; route?: { ...; } | undefined; headers?: { ...; } | undefined; }' is not assignable to parameter of type 'P'.
  '{ query: Query; adapter?: Partial<AdapterServiceOptions> | undefined; paginate?: PaginationParams | undefined; provider?: string | undefined; route?: { ...; } | undefined; headers?: { ...; } | undefined; }' is assignable to the constraint of type 'P', but 'P' could be instantiated with a different subtype of constraint 'AdapterParams<AdapterQuery, Partial<AdapterServiceOptions>>'.deno-ts(2345)

I don't know if there is a way to get around this without type casting (what would of course be very cool).

Btw we could just cast `params` as `P`, but I don't know if it is more readable:
```ts
    return this.$get(id, {
      ...(params as P),
      query,
    });
```